src/app.service.ts

```diff
@@ -9,72 +9,67 @@ import { date, number, string } from 'joi';
 import axios from 'axios';
 import { config } from 'rxjs';
 import Joi from "joi";
+import { Session } from 'express-session';
 
 @Injectable()
 export class AppService {
-  public transactions: TransactionModel[] =[]
-  public payUnitData={
-    transaction_id:10,
-    transaction_amount:10000,
-    currency:'USD',
-    return_url:'http://localhost:3000',
-  };
-  // private all
-  private completeTransaction = {
-    gateway: 'mtnmomo',
-    transaction_amount: 1001,
-    transaction_id: 1005,
-    phone_number: 681479697,
-    currency: 'USD',
-    paymentType: 'button',
-  };
+  private all
+  public  getIninitial: GetTransaction[] =[]
+  public transaction: TransactionModel[] =[]
+  public  payment:Payment[] =[]
   private urlinitialized = 'https://app.payunit.net/api/gateway/initialize';
   private paymentUrl = 'https://app.payunit.net/api/gateway/makepayment';
-  private urlproceed = 'https://app.payunit.net/api';
+  private urlproceed = 'https://app.payunit.net/api/gateway/gateways?t_url=""&t_id=""&t_sum=""';
   constructor(
     private httpService: HttpService,
     private configService: ConfigService,
   ) {}
 
-  async postApi() {
+  async postApi(transaction_id:string, total_amount:number,  currency:string, return_url:string) {
+    const newTransaction =new TransactionModel(
+      transaction_id,
+      total_amount,
+      currency,
+      return_url)
+    this.transaction.push(newTransaction)
     const appBasic = `${process.env.apiSecret}`;
     const buff = Buffer.from(appBasic);
     const base64data = buff.toString('base64');
     console.log(base64data);
```

Comment:
Since you will always use these header variables in all API calls, write a function to initialize the headers which you will just call where ever you need in.
Write an interceptor that will intercept all requests you will be making to Payunit and append the header to it in order for you to be able to make the call.